### ISSUE TITLE
Neither ballerina/jms nor ballerina/mb seem able to connect to RabbitMQ

### ISSUE THREAD

**Issue by autodidaddict** [ballerina-platform/ballerina-lang#9746]
**Description:**
RabbitMQ supports AMQP 0.9.1 and the Ballerina documentation claims that the Ballerina message broker is also AMQP, therefore one would assume that either the `mb` or the `jms` package should be able to talk to rabbit.

**Steps to reproduce:**
Use the following code:
```
import ballerina/mb;
import ballerina/log;

documentation { Queue receiver endpoint for drone location updates }
endpoint mb:SimpleQueueReceiver queueReceiverDroneLocations {
    host: "localhost",
    port: 5672,
    username: "guest",
    password: "guest",
    queueName: "DroneLocations"
};

service<mb:Consumer> locationListener bind queueReceiverDroneLocations {
    onMessage(endpoint consumer, mb:Message message) {
        string messageText = check message.getTextMessageContent();
        log:printInfo("[DONE LOCATION] " + messageText);
    }
}
```
Run a local RabbitMQ instance with all default settings _and_ enable the JMS topic plugin.
Error message is as follows:
```
ballerina: initiating service(s) in 'drone_loc_monitor.bal'
error: ballerina/runtime:CallFailedException, message: call failed
	at ..<init>(drone_loc_monitor.bal:5)
caused by ballerina/runtime:CallFailedException, message: call failed
	at ballerina/mb:init(simple_queue_receiver.bal:38)
caused by ballerina/runtime:CallFailedException, message: call failed
	at ballerina/jms:init(simple_queue_receiver.bal:39)
caused by error, message: error while connecting to broker. Error creating connection: Unable to Connect
	at ballerina/jms:new(connection.bal:27)
```
I have tried to connect to RabbitMQ with both the simple `jms` receiver and the simple `mb` receiver, neither work properly and both result in various error messages indicating failure. If there is a specific set of properties required to speak "pure" AMQP, then it's not discernible from the documentation.

**Affected Versions:**
Ballerina 0.980.1

**OS, DB, other environment details and versions:**
Ubuntu 18.04, MacOS High Sierra, RabbitMQ 3.6.10 (Erlang 20.2.2)

**Suggested Labels (optional):**
documentation, tutorial, bug

I understand you're working on an AMQP connector. In the meantime, it would help if what you can and cannot do with the existing connectors was made more clear and, if RabbitMQ is possible, then instructions on how to configure it should be made available.

**Comment by autodidaddict** [comment]
I said in the issue that it doesn't with with jms either. I have not been
able to find a configuration that works. Can you please show an example of
this working with rabbit before closing the issue?

**Comment by riyafa** [comment]
Created a blog to help you with: https://riyafa.wordpress.com/2018/08/08/how-to-configure-ballerina-jms-to-work-with-rabbitmq/
You hadn't mentioned how you tried with JMS. It is all about configuring a JMS application to work with RabbitMQ.

**Comment by a5anka** [comment]
@autodidaddict `mb` endpoint is designed as an specific connector which can be used to connect with BMB. Please note that it is not meant to be used as an generic AMQP connector. Both the RabbitMQ and BMB is not 100% AMQP spec complient. Therefore the `mb` connector might not work with RabbitMQ OOB. I also think that we should not make the `mb`  endpoint work with every AMQP broker because in the future we might use a different protocol to talk to BMB. Alternatively as you have mentioned what we should do is implementing a generic AMQP connector which you can use to connect to a broker using AMQP. WDYT?

**Comment by autodidaddict** [comment]
Absolutely. Sorry I wasn't clear. I think the `mb` endpoint should stay focused on the Ballerina-proprietary broker (but perhaps the documentation could be more clear that it _only_ works with that broker), and then having a separate `amqp` broker that works with Rabbit and Rabbit-likes would be fantastic. Ideally I think we'd all love to see an `mqtt` endpoint in the future, too, but hopefully things like that will be picked up once the community grows.

**Comment by a5anka** [comment]
Thanks, @autodidaddict for the feedback. We will work on improving the documentation.

**Comment by Manuri** [comment]
I have created separate issues to track the [AMQP client implementation](https://github.com/ballerina-platform/ballerina-lang/issues/10115) and [documentation improvement](https://github.com/ballerina-platform/ballerina-lang/issues/10116).

Hence closing this issue.

**Comment by wggihan** [comment]
@autodidaddict Just want to let you know that, we have implemented a RabbitMQ consumer, producer for Ballerina and that is package with default distribution. This is now available with ballerina-1.0.0-alpha pack.
